common/test/acceptance/pages/studio/settings_advanced.py

```diff
@@ -34,6 +37,36 @@ def _is_finished_loading():
         EmptyPromise(_is_finished_loading, 'Finished rendering the advanced policy items.').fulfill()
         return self.q(css='body.advanced').present
 
+    def get_key_names(self):
+        """
+            Returns a list of key names of all settings.
+        """
+        return self.q(css=KEY_CSS).text
+
+    def get_deprecated_settings_button_text(self):
+        """
+            Returns text for deprecated settings button
+        """
+        return self.q(css=DEPRECATED_SETTINGS_BUTTON_SELECTOR).text[0]
+
+    def click_deprecated_settings_button(self):
+        """
+            Clicks deprecated settings button
+        """
+        self.q(css=DEPRECATED_SETTINGS_BUTTON_SELECTOR).click()
```

Comment:
Instead of "click button" you should have something more stateful, like "go to deprecated settings", and then, after the click, you should have a `wait` statement of some kind. I see in this PR that you have click, then immediately assert. That is going to lead to flaky tests. Better to have a wait in there to ensure the state of the page has met your needs before asserting.